secure/storage/src/namespaced_storage.rs

```diff
@@ -1,42 +1,44 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{CryptoKVStorage, Error, GetResponse, KVStorage, Value};
+use crate::{CryptoKVStorage, Error, GetResponse, KVStorage, Storage};
+use serde::{de::DeserializeOwned, Serialize};
+use std::sync::{Arc, RwLock};
 
 /// This provides a light wrapper around KV storages to support a namespace. That namespace is
 /// effectively prefixing all keys with then namespace value and "/" so a namespace of foo and a
 /// key of bar becomes "foo/bar". Without a namespace, the key would just be "bar". This matches
 /// how this library implements namespaces for Vault.
 pub struct NamespacedStorage {
     namespace: String,
-    inner: Box<dyn KVStorage>,
+    inner: Arc<RwLock<Storage>>,
```

Comment:
why

Reply:
Oh you know, I think I can use a Box<Storage>... I think I made this to convoluted, good call out.